### PR TITLE
Bundles

### PR DESCRIPTION
**Pull request by timmillwood**

### CODE REVIEW

File: multiversion.module [hunk 1]
```diff
@@ -348,3 +351,24 @@ function multiversion_form_alter(&$form, \Drupal\Core\Form\FormStateInterface $f
...
+
+function template_preprocess_workspace_add_list(&$variables) {
+  if (!empty($variables['content'])) {
+    foreach ($variables['content'] as $type) {
+      $variables['types'][$type->id()]['label'] = $type->label();
+      $options = array('query' => \Drupal::request()->query->all());
+      $variables['types'][$type->id()]['url'] = \Drupal::url('entity.workspace.add_form', array('workspace_type' => $type->id()), $options);
+    }
+  }
+}
```
Comment (jeqq) [inline multiversion.module]: Here should be a new line :)

File: src/Form/WorkspaceTypeDeleteForm.php [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      $caption = '<p>' . $this->formatPlural(count($workspaces), '%label is used by 1 workspace on your site. You can not remove this workspace type until you have removed all of the %label workspaces.', '%label is used by @count workspaces on your site. You may not remove %label until you have removed all of the %label custom workspaces.', array('%label' => $this->entity->label())) . '</p>';
+      $form['description'] = array('#markup' => $caption);
+      return $form;
+    }
+    else {
+      return parent::buildForm($form, $form_state);
+    }
+  }
+
+}
```
Comment (jeqq) [inline src/Form/WorkspaceTypeDeleteForm.php]: New line missing

File: src/WorkspaceListBuilder.php [hunk 3]
```diff
@@ -7,6 +7,7 @@
 
 namespace Drupal\multiversion;
 
+
```
Comment (jeqq) [inline src/WorkspaceListBuilder.php]: Here is an extra new line

File: templates/workspace-add-list.html.twig [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+ * @ingroup themeable
+ */
+#}
+<ul class="admin-list">
+    {% for type in types %}
+        <li class="clearfix">
+            <a href="{{ type.url }}"><span class="label">{{ type.label }}</span></a>
+        </li>
+    {% endfor %}
+</ul>
```
Comment (jeqq) [inline templates/workspace-add-list.html.twig]: new line

File: multiversion.install [hunk 5]
```diff
@@ -1,13 +1,16 @@
...
+use Drupal\multiversion\Entity\WorkspaceType;
 
 /**
  * Implements hook_install().
  */
 function multiversion_install() {
   // Create default workspace.
-  Workspace::create(['machine_name' => 'default', 'label' => 'Default'])->save();
+  $basic = WorkspaceType::create(['id' => 'basic', 'label' => 'Basic workspace']);
+  $basic->save();
```
Comment (dickolsson) [inline multiversion.install]: Let's move this to a config yml file instead.

**Pull request merged**